packages/graphback-datasync/src/providers/DataSyncConflictProvider.ts

```diff
@@ -0,0 +1,126 @@
+import { ModelDefinition, GraphbackContext, TransformType, FieldTransform, NoDataError, GraphbackOperationType } from '@graphback/core';
+import { DataSyncModelConflictConfig, DataSyncFieldNames, ConflictMetadata, ConflictError } from '../util';
+import { MongoDeltaHelper } from '../deltaHelper';
+import { DataSyncMongoDBDataProvider } from './DatasyncMongoDBDataProvider';
+
+export const MAX_RETRIES = 3;
+
+/**
+ * Da
+ */
+export class DataSyncConflictMongoDBDataProvider<Type = any> extends DataSyncMongoDBDataProvider<Type> {
+  protected conflictConfig: DataSyncModelConflictConfig
+  protected deltaHelper: MongoDeltaHelper
+
+  public constructor(model: ModelDefinition, client: any, dataSyncConflictConfig: DataSyncModelConflictConfig) {
+    super(model, client);
+    this.conflictConfig = dataSyncConflictConfig;
+
+    if (!this.conflictConfig.conflictResolution?.resolveUpdate) {
+      this.conflictConfig.conflictResolution = {
+        resolveUpdate(conflict: ConflictMetadata) {
+          throw new ConflictError(conflict)
+        }
+      }
+    }
+
+    this.deltaHelper = new MongoDeltaHelper(model, client);
+
+  }
+
+  public async create(data: any, context: GraphbackContext): Promise<Type> {
+    const contextWithoutSelectedFields: GraphbackContext = { ...context }
+    contextWithoutSelectedFields.graphback.options.selectedFields = [];
```

Comment:
Why is this needed?

Reply:
@machi1990 @craicoverflow @wtrocki I am using it to get the complete document instead of just the selected fields. This is to ensure that the complete document is stored in the delta table:
https://github.com/aerogear/graphback/blob/601fe9f40cadf9ee5617073f491d05052d6ed9e5/packages/graphback-datasync/src/providers/DataSyncConflictProvider.ts#L37-L39

Perhaps we do not need a delta table entry for the `create` operation, wdyt :thinking:

Reply:
Yeah you are right to do this - I was thinking that all fields are fetched anyway, but only if subscriptions are enabled for that mutation type.

However, I think you can just modify the context directly, instead of cloning it to remove the selectedFields. There should be one context per request, and it has a short lifetime so modifying it directly should be fine.

Reply:
See https://github.com/aerogear/graphback/blob/master/packages/graphback-core/src/runtime/CRUDService.ts#L43